tests/integration/testlib/plugin.py

```diff
@@ -0,0 +1,46 @@
+#
+# Copyright (c) 2020 Red Hat, Inc.
+#
+# This file is part of nmstate
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation, either version 2.1 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+#
+
+from contextlib import contextmanager
+import re
+
+from . import cmdlib
+
+
+SYSTEMCTL_RESTART_NM_CMD = ("systemctl", "restart", "NetworkManager")
+
+
+@contextmanager
+def disable_nm_plugin(plugin):
+    ret, rpm_output, _ = cmdlib.exec_cmd(
+        ("rpm", "-ql", f"NetworkManager-{plugin}")
+    )
+    if not ret:
+        lib_path = re.search(r"[A-Za-z0-9-/.]*.so", rpm_output).group()
+        cmdlib.exec_cmd(("mount", "--bind", "/dev/null", lib_path))
+        cmdlib.exec_cmd(SYSTEMCTL_RESTART_NM_CMD)
+        plugin_disable = True
+    else:
+        plugin_disable = False
```

Comment:
Can't you just exit earlier if ret evaluates to True? I see no need for the flags in that case.
```
if ret:
    return
```

Reply:
That is not possible. If we do this and the user does not have the plugin installed the test is going to fail with the following error:
```
    def test_nm_ovs_plugin_missing():
>       with disable_nm_plugin("ovs"):

tests/integration/ovs_test.py:151: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <contextlib._GeneratorContextManager object at 0x7ff54c4e9a60>

    def __enter__(self):
        # do not keep args and kwds alive unnecessarily
        # they are only needed for recreation, which is not possible anymore
        del self.args, self.kwds, self.func
        try:
            return next(self.gen)
        except StopIteration:
>           raise RuntimeError("generator didn't yield") from None
E           RuntimeError: generator didn't yield
```